.github/workflows/ci.yaml

```diff
@@ -0,0 +1,121 @@
+name: CI Pipeline
+
+on:
+  workflow_dispatch:
+    inputs:
+      docker_registry:
+        description: Docker registry
+        required: true
+        default: ghcr.io
+        type: string
+      docker_repo:
+        description: Docker repository name
+        required: true
+        default: angular-react-starter-image
+        type: string
+      docker_tag:
+        description: Docker tag
+        required: true
+        default: latest
+        type: string
+      mail_server_address:
+        description: Mail server address
+        required: true
+        default: smtp.gmail.com
+        type: string
+      mail_server_port:
+        description: Mail server port
+        required: true
+        default: 465
+        type: number
+      mail_receiver: 
+        description: Email of the receiver
+        required: false
+        type: string
+    secrets:
+      mail_username:
+        description: Username for the email notification step
+        required: true
+      mail_password:
+        description: Password for the email notification step
+        required: true
+      github_token:
+        description: GitHub token for accessing the GitHub container registry
+        required: true
+
+env:
+  IMAGE_NAME: '${{ inputs.docker_registry }}/${{ github.actor }}/${{ inputs.docker_repo }}:${{ inputs.docker_tag }}'
+  DEFAULT_MAIL_RECEIVER: '${{ github.actor_id }}+${{ github.actor }}@users.noreply.github.com'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./angular
+    steps:
+      - name: Checkout code from the repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+
+      - name: Install clean dependencies
+        run: npm ci
+
+      - name: Run headless tests
+        id: run-tests
+        run: npm run test -- --watch=false --browsers=ChromeHeadless 
+
+      - name: Build Angular project
+        id: build-project
+        run: npm run build
+
+      - name: Log in to GitHub container registry  
+        id: login-registry
+        run: | 
+          echo "${{ secrets.github_token }}" | docker login \
+          ${{ inputs.docker_registry }} -u ${{ github.actor }} \
+          --password-stdin
+
+      - name: Build Docker image
+        id: build-docker
+        run: docker build -t $IMAGE_NAME .
+        working-directory: .
+
+      - name: Push Docker image
+        id: push-docker
+        run: docker push $IMAGE_NAME
+
+      - name: Get pipeline execution time
+        id: get-time
+        if: always()
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          start_time=$(gh run list --json createdAt -w ci.yaml -L 1 | jq -r '.[0].createdAt')
+          end_time=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
```

Comment:
use updatedAt, of the gh output to get end_time

Reply:
I'm not sure if I understood you correctly, but I changed `end_time` to use `updatedAt` instead of the current time. The problem is that `updatedAt` is supposed to be the time of the last workflow status update, but it's lagging behind the current time

Reply:
so they update the status and then set the updateTime, so we can use it to calculate the duration. It's enough for checking the speed of the pipeline.